test/abstractarray.jl

```diff
@@ -682,6 +682,9 @@ function test_cat(::Type{TestAbstractArray})
     @test Base.typed_hcat(Float64, B, B) == TSlow(b2hcat)
     @test Base.typed_hcat(Float64, B, B, B) == TSlow(b3hcat)
 
+    # issue #49676
+    @test_throws ArgumentError Base.typed_hcat([1 2 3], 1, +1)
```

Comment:
Probably good to add another test with `vec[1 +1]` syntax so it's clear why this throws an ArgumentError

Reply:
Replaced the test with something with the v[1 +1] syntax and a comment explaining why it's next to the typed_hcat tests.